src/lib/util/slab.h

```diff
@@ -160,34 +167,29 @@ extern "C" {
 	static inline CC_HINT(nonnull(2)) int fr_ ## _name ## _slab_list_alloc(TALLOC_CTX *ctx, \
 									       fr_ ## _name ## _slab_list_t **out, \
 									       fr_event_list_t *el, \
-									       fr_time_delta_t interval, \
-									       unsigned int elements_per_slab, \
-									       unsigned int min_elements, \
-									       unsigned int max_elements, \
-									       bool at_max_fail, \
-									       size_t num_children, \
-									       size_t child_pool_size, \
+									       fr_slab_config_t const *config, \
 									       fr_ ## _type ## _slab_alloc_t alloc, \
 									       fr_ ## _type ## _slab_reserve_t reserve, \
 									       void *uctx, \
 									       bool release_reset) \
 	{ \
 		MEM(*out = talloc_zero(ctx, fr_ ## _name ## _slab_list_t)); \
 		(*out)->el = el; \
-		(*out)->interval = interval; \
-		(*out)->elements_per_slab = elements_per_slab; \
-		(*out)->min_elements = min_elements; \
-		(*out)->max_elements = max_elements; \
-		(*out)->at_max_fail = at_max_fail; \
+		(*out)->interval = config->interval; \
+		(*out)->elements_per_slab = (config->elements_per_slab > 0 ? config->elements_per_slab : \
+					     (config->min_elements ? config->min_elements : 1)); \
+		(*out)->min_elements = config->min_elements; \
+		(*out)->max_elements = config->max_elements; \
+		(*out)->at_max_fail = config->at_max_fail; \
 		(*out)->alloc = alloc; \
-		(*out)->num_children = num_children; \
-		(*out)->child_pool_size = child_pool_size; \
+		(*out)->num_children = config->num_children; \
+		(*out)->child_pool_size = config->child_pool_size; \
```

Comment:
Just use the config internally too in the `_slab_list_t` then we don't have to add more boilerplate every time we add a config item :P